### PR TITLE
Fix

### PR DESCRIPTION
**Pull request by CoderJoeW**
Correct database intergration(wont interrupt game if one is not present)

Includes code to log incorrect passwords

Theres probably some other stuff i added in here but i cant remember(i mean it took 4 months for the first commit to be accepted)
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/huytd/agar.io-clone/pull/490%23issuecomment-290332701%22%2C%20%22https%3A//github.com/huytd/agar.io-clone/pull/490%23issuecomment-290348640%22%2C%20%22https%3A//github.com/huytd/agar.io-clone/pull/490%23issuecomment-290354208%22%2C%20%22https%3A//github.com/huytd/agar.io-clone/pull/490%23issuecomment-290607119%22%2C%20%22https%3A//github.com/huytd/agar.io-clone/pull/490%23discussion_r109258582%22%2C%20%22https%3A//github.com/huytd/agar.io-clone/pull/490%23issuecomment-291009799%22%2C%20%22https%3A//github.com/huytd/agar.io-clone/pull/490%23discussion_r109345228%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/huytd/agar.io-clone/pull/490%23issuecomment-290332701%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Please%20fix%20merge%20conflicts%5Cr%5Cn%5Cr%5CnAlso%2C%20you%20appear%20to%20have%20added%20your%20AWS%20credentials%20to%20config.json%2C%20probably%20should%20remove%20those%22%2C%20%22created_at%22%3A%20%222017-03-30T07%3A59%3A17Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars1.githubusercontent.com/u/2415829%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/abalabahaha%22%7D%7D%2C%20%7B%22body%22%3A%20%22And%20please%20for%20god%27s%20sake%2C%20squash%20it.%22%2C%20%22created_at%22%3A%20%222017-03-30T09%3A04%3A17Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars3.githubusercontent.com/u/16986290%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Nikitaw99%22%7D%7D%2C%20%7B%22body%22%3A%20%22Fixed%20conflicts//Added%20comments%20to%20config//%20Removed%20my%20auth%20vals%22%2C%20%22created_at%22%3A%20%222017-03-30T09%3A26%3A58Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars0.githubusercontent.com/u/14892313%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/flaminggenius%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3Ashipit%3A%22%2C%20%22created_at%22%3A%20%222017-03-31T03%3A35%3A30Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars0.githubusercontent.com/u/14892313%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/flaminggenius%22%7D%7D%2C%20%7B%22body%22%3A%20%22fixed%22%2C%20%22created_at%22%3A%20%222017-04-02T19%3A43%3A29Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars0.githubusercontent.com/u/14892313%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/flaminggenius%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%201720a17ab32700ce048b7aabbd194a5040f24f10%20config.json%205%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/huytd/agar.io-clone/pull/490%23discussion_r109258582%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Comments%20are%20not%20valid%20JSON%2C%20please%20remove%20them%22%2C%20%22created_at%22%3A%20%222017-03-31T22%3A03%3A33Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars1.githubusercontent.com/u/2415829%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/abalabahaha%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20config.json%3AL32-44%22%7D%2C%20%22Pull%20e6d5d36a292482f8e889dd60297317c0898a0112%20src/server/server.js%2010%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/huytd/agar.io-clone/pull/490%23discussion_r109345228%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22You%20should%20be%20using%20the%20config%20option%20here%2C%20not%20an%20environment%20variable%22%2C%20%22created_at%22%3A%20%222017-04-03T06%3A14%3A11Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars1.githubusercontent.com/u/2415829%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/abalabahaha%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20src/server/server.js%3AL6-25%22%7D%7D%2C%20%22approved%22%3A%20%7B%22https%3A//github.com/flaminggenius%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars0.githubusercontent.com/u/14892313%3Fv%3D3%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>

<a href='https://github.com/flaminggenius'></a>

- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/huytd/agar.io-clone/pull/490#issuecomment-290332701'>General Comment</a></b>
- <a href='https://github.com/abalabahaha'></a> Please fix merge conflicts
Also, you appear to have added your AWS credentials to config.json, probably should remove those
- <a href='https://github.com/Nikitaw99'></a> And please for god's sake, squash it.
- <a href='https://github.com/flaminggenius'></a> Fixed conflicts//Added comments to config// Removed my auth vals
- <a href='https://github.com/flaminggenius'></a> :shipit:
- <a href='https://github.com/flaminggenius'></a> fixed
- [ ] <a href='#crh-comment-Pull 1720a17ab32700ce048b7aabbd194a5040f24f10 config.json 5'></a> &nbsp;<b><a href='https://github.com/huytd/agar.io-clone/pull/490#discussion_r109258582'>File: config.json:L32-44</a></b>
- <a href='https://github.com/abalabahaha'></a> Comments are not valid JSON, please remove them
- [ ] <a href='#crh-comment-Pull e6d5d36a292482f8e889dd60297317c0898a0112 src/server/server.js 10'></a> &nbsp;<b><a href='https://github.com/huytd/agar.io-clone/pull/490#discussion_r109345228'>File: src/server/server.js:L6-25</a></b>
- <a href='https://github.com/abalabahaha'></a> You should be using the config option here, not an environment variable

<a href='https://www.codereviewhub.com/huytd/agar.io-clone/pull/490?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/huytd/agar.io-clone/pull/490?approve=1'></a>&nbsp;<a href='https://www.codereviewhub.com/huytd/agar.io-clone/pull/490?approve=0'></a>&nbsp;<a href='https://github.com/huytd/agar.io-clone/pull/490'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by huytd** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/huytd/agar.io-clone/pull/490'>Refresh</a>.<a href='#crh-update-none'></a>

**Comment by abalabahaha** [comment]
Please fix merge conflicts

Also, you appear to have added your AWS credentials to config.json, probably should remove those

**Comment by Nikitaw99** [comment]
And please for god's sake, squash it.

**Comment by flaminggenius** [comment]
Fixed conflicts//Added comments to config// Removed my auth vals

**Comment by flaminggenius** [comment]
:shipit:

File: config.json [hunk 1]
```diff
@@ -32,11 +32,12 @@
     "minMassLoss": 50,
     "mergeTimer": 15,
     "sqlinfo":{
-      "connectionLimit": 100,
+      "connectionLimit": 100,//Higher values may cause lag
```
Comment (abalabahaha) [inline config.json]: Comments are not valid JSON, please remove them

**Comment by flaminggenius** [comment]
fixed

File: package.json [hunk 2]
```diff
@@ -59,6 +59,8 @@
     "sync-request": "^3.0.1",
     "webpack": "^1.13.1",
     "webpack-stream": "^3.2.0",
-    "mysql": "^2.5.5"
+    "mysql": "^2.5.5",
```
Comment (abalabahaha) [inline package.json]: You're not using this module anymore

File: package.json [hunk 3]
```diff
@@ -59,6 +59,8 @@
...
+    "pg": "^6.1.0",
+    "node-localstorage": "^1.3.0"
```
Comment (abalabahaha) [inline package.json]: You're not using this anywhere

File: src/server/server.js [hunk 4]
```diff
@@ -6,7 +6,19 @@ var app = express();
 var http = require('http').Server(app);
 var io = require('socket.io')(http);
 var SAT = require('sat');
-var sql = require ("mysql");
+var pg = require ('pg');
+var localStorage = require ('node-localstorage');
+
+//Establish database connection
+pg.defaults.ssl = true;
```
Comment (abalabahaha) [inline src/server/server.js]: This seems quite arbitrary, why not make it a config option?

File: src/server/server.js [hunk 5]
```diff
@@ -6,7 +6,19 @@ var app = express();
...
+var client = new pg.Client(process.env.DATABASE_URL);
```
Comment (abalabahaha) [inline src/server/server.js]: You should be using the config option here, not an environment variable

File: src/server/server.js [hunk 6]
```diff
@@ -6,7 +6,19 @@ var app = express();
...
+client.connect (function (err){
```
Comment (abalabahaha) [inline src/server/server.js]: This still warns users even if they don't want to use a database.

Maybe set the default `config.sqlinfo.host` to an empty string, and only attempt to connect if the config host is not nothing

**Comment by flaminggenius** [comment]
Resolved conflicts now ready to be deployed!!

**Comment by flaminggenius** [comment]
Needs to be merged to fix database issues introduced in last pull request

**Comment by huytd** [comment]
And let's [squash](https://github.com/huytd/agar.io-clone/pull/471#issuecomment-328596683) this pull request into a single commit :D

**Comment by flaminggenius** [comment]
When I find some free time I will work on adding a level up system perhaps based on a timeAlive/avgMass?

File: src/server/server.js (hunk 4 above)
Comment (3642066) [reply src/server/server.js]: Agreed !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

**Pull request closed without merging**